workflows/methylation/methylation-cohort.wdl

```diff
@@ -0,0 +1,245 @@
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        ),
+    }
+
+    call generate_umap { input:
+        filtered_beta_values = filter_probes.filtered_beta_values,
+    }
+
+    call plot_umap { input:
+        umap = generate_umap.umap,
+    }
+
+    output {
+        File combined_beta = select_first(
+            [
+                final_merge.combined_beta,
+                simple_merge.combined_beta,
+            ]
+        )
+        File filtered_beta = filter_probes.filtered_beta_values
+        File filtered_probeset = filter_probes.filtered_probes
+        File umap_embedding = generate_umap.umap
+        File umap_plot = plot_umap.umap_plot
+    }
+}
+
+task combine_data {
+    meta {
+        description: "Combine data from multiple samples"
+        outputs: {
+            combined_beta: "Combined beta values for all samples"
+        }
+    }
+
+    parameter_meta {
+        unfiltered_normalized_beta: "Array of unfiltered normalized beta values for each sample"
+        combined_file_name: "Name of the combined file"
+        modify_memory_gb: "Add to or subtract from dynamic memory allocation. Default memory is determined by the size of the inputs. Specified in GB."
+    }
+
+    input {
+        Array[File] unfiltered_normalized_beta
+        String combined_file_name = "combined_beta.csv"
+        Int modify_memory_gb = 0
+    }
+
+    Int memory_gb = ceil(size(unfiltered_normalized_beta, "GiB")) + modify_memory_gb
+    Int disk_size_gb = ceil(size(unfiltered_normalized_beta, "GiB") * 2)
+
+    command <<<
+        echo "Combining data"
+
+        python $(which combine.py) --output-name ~{combined_file_name} ~{sep(" ", unfiltered_normalized_beta)}
+    >>>
+
+    output {
+        File combined_beta = combined_file_name
+    }
+
+    runtime {
+        container: "ghcr.io/stjudecloud/pandas:branch-methylation-2.2.1-0"
+        memory: "~{memory_gb} GB"
+        cpu: 1
+        disks: "~{disk_size_gb} GB"
+        maxRetries: 1
+    }
+}
+
+task filter_probes {
+    meta {
+        description: "Filter probes based on standard deviation"
+        help: "Probes are filtered by calculating the standard deviation of the beta value for each probe across all samples. The top N probes with the highest standard deviation are retained."
+        outputs: {
+            filtered_beta_values: "Filtered beta values for all samples",
+            filtered_probes: "Probes that were retained after filtering.",
+        }
+    }
+
+    parameter_meta {
+        beta_values: "Beta values for all samples"
+        num_probes: "Number of probes to retain after filtering"
+    }
+
+    input {
+        File beta_values
+        Int num_probes = 10000
+    }
+
+    Int disk_size_gb = ceil(size(beta_values, "GiB") * 2)
+
+    #@ except: LineWidth
+    command <<<
+        ln -s ~{beta_values} beta.csv
+
+        python $(which filter.py) --num_probes ~{num_probes} beta.csv
```

Comment:
```suggestion
        python $(which filter.py) --num_probes ~{num_probes} ~{beta_values}
```
With my request to add argparse this can be a bit cleaner (and avoids adding `set -euo pipefail`)